cpp/src/arrow/compute/kernels/test_util_test.cc

```diff
@@ -0,0 +1,38 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include "arrow/compute/kernels/test_util.h"
+#include "arrow/testing/gtest_util.h"
+
+namespace arrow::compute {
+
+TEST(RunEndEncodeGtestUtilTest, SchemaTypeIsModified) {
```

Comment:
Rename this test to something nicer than `RunEndEncodeGtestUtilTest`?

Reply:
What about?
```suggestion
TEST(RunEndEncodeTableColumnsUtilTest, SchemaTypeIsModified) {
```

Reply:
How about `RunEndEncodeTableColumnsTest`?